source/ui/memory-component-presenter.cpp

```diff
@@ -50,18 +47,18 @@ void MemoryComponentPresenter::onMemoryChanged(std::size_t address,
 
   // if the memory that is hold in cache is changed, invalidate cache
   // new value intersects with the beginning of the cached region
-  bool overlap_beginning =
+  bool overlapBeginning =
```

Comment:
Couldn't we reduce this to `(address >= begin && address <= end) || ((address + length) >= begin && (address + length) <= end)`. We just care about an overlap anywhere